packages/design-system/src/new-ui/DataTable/DataTable.tsx

```diff
@@ -28,8 +29,12 @@ interface DataTableProps<TData, TValue> {
   isLoading: boolean;
   loadingRows: Nullable<number>;
   children?: React.ReactNode;
-  primaryText: Nullable<string>;
-  secondaryText: Nullable<string>;
+  primaryText?: Nullable<string>;
+  secondaryText?: Nullable<string>;
+  manualPagination?: boolean;
+  pageCount?: number;
+  onPaginationChange?: (state: PaginationState) => void;
+  pagination?: PaginationState;
```

Comment:
@orangecoloured I will recommend we use paginationState and onPaginationStateChange.

In this codebase, we believe the exchange between longer context with better readability is beneficial. So don't be afraid to have long variable name. Most of time it is very helpful for maintainer